src/main/java/br/com/coderbank/redfoxghs/account_api/services/AccountService.java

```diff
@@ -26,32 +27,62 @@ public AccountResponseDTO create(UUID idClient) {
-        accountEntity.setIdClient(idClient);
+        AccountEntity accountEntity = new AccountEntity(idClient);
 
         try {
             AccountEntity savedAccount = accountRepository.save(accountEntity);
-            return new AccountResponseDTO(
-                    savedAccount.getIdAccount(),
-                    savedAccount.getIdClient(),
-                    savedAccount.getAgencyNumber(),
-                    savedAccount.getAccountNumber(),
-                    savedAccount.getBalance()
-            );
+            return new AccountResponseDTO(savedAccount);
         } catch (DataIntegrityViolationException e) {
             throw new CustomDatabaseException("Erro de integridade de dados ao salvar a conta", e);
-        } catch (Exception e) {
-            throw new GeneralDatabaseException("Erro desconhecido ao salvar a conta", e);
         }
     }
 
     public AccountBalanceResponseDTO getBalanceAccount(UUID idAccount) {
+        AccountEntity accountEntity = accountRepository.findById(idAccount)
+                .orElseThrow(() -> new NotFoundDatabaseException("Não foi encontrada uma conta com esse id: " + idAccount));
 
-        Optional<AccountEntity> accountEntityOptional = Optional.ofNullable(accountRepository.findById(idAccount)
-                .orElseThrow(() -> new NotFoundDatabaseException("Não foi encontrada uma conta com esse id " + idAccount)));
+        return new AccountBalanceResponseDTO(accountEntity.getBalance());
+    }
+
+    public void increaseAccountBalance(AccountBalanceRequestDTO increaseAccountBalance) {
+        UUID accountId = UUID.fromString(increaseAccountBalance.idAccount());
+        BigDecimal amountToAdd = validateAndGetAmount(increaseAccountBalance);
+
+        AccountEntity accountEntity = this.findAccount(accountId);
+
+        accountEntity.addBalance(amountToAdd);
+
+        accountRepository.save(accountEntity);
+    }
+
+    public void decreaseAccountBalance(AccountBalanceRequestDTO decreaseAccountBalance) {
+        UUID accountId = UUID.fromString(decreaseAccountBalance.idAccount());
+        BigDecimal amountToSubtract = validateAndGetAmount(decreaseAccountBalance);
+
+        AccountEntity accountEntity = this.findAccount(accountId);
+
+        validateSufficientBalance(accountEntity, amountToSubtract);
 
-        return new AccountBalanceResponseDTO(
-                accountEntityOptional.get().getBalance()
-        );
+        accountEntity.subtractBalance(amountToSubtract);
+
+        accountRepository.save(accountEntity);
+    }
+
+    private AccountEntity findAccount(UUID accountId) {
+        return accountRepository.findById(accountId)
+                .orElseThrow(() -> new NotFoundDatabaseException("Não foi encontrada uma conta com esse id: " + accountId));
+    }
+
+    private BigDecimal validateAndGetAmount(AccountBalanceRequestDTO request) {
+        BigDecimal amount = request.balance();
+        if (amount.signum() <= 0) {
```

Comment:
Não cheguei a testar o código, mas me parece que não deveria ser necessário essa validação aqui devido no AccountBalanceRequestDTO ter a validação a baixo que é responsável por não permitir valores negativos como entrada. 

        @Positive(message = "The value must be greater than zero.")
        BigDecimal balance